### PR TITLE
address TODOs

### PR DESCRIPTION
**Pull request by universenox**

### CODE REVIEW

File: include/real/exact_number.hpp [hunk 1]
```diff
@@ -1222,8 +1214,11 @@ namespace boost {
                 if (exponent < 0) {
                     return false;
                 } else {
-                    /// @TODO: ensure it doesn't overflow
-                    return (exponent_t) digits.size() <= exponent;
+                    if (exponent < 0)
+                        return false;
```
Comment (sdavtaker) [inline include/real/exact_number.hpp]: Isn't this the same check 2 lines above?

File: include/real/exact_number.hpp [hunk 2]
```diff
@@ -1222,8 +1214,11 @@ namespace boost {
...
+                    else {
+                        return digits.size() <= (size_t) exponent;
```
Comment (sdavtaker) [inline include/real/exact_number.hpp]: This is in the territory of undefined behavior:

Example:
```
    int i = -1;
    std::size_t s = (std::size_t) i;
    std::cout << s << std::endl;
```

outputs: 18446744073709551615

Comment (universenox) [reply include/real/exact_number.hpp]: That is why I return false if exponent < 0. At that point, exponent is always >= 0 so it should not have problems when cast

Comment (sdavtaker) [reply include/real/exact_number.hpp]: But why you check twice?
```
if (exponent < 0) {
                    return false;
                } else {
                    if (exponent < 0)
                        return false;
...
```

It seems like the second if is deadcode

Comment (universenox) [reply include/real/exact_number.hpp]: Oops. I'll remove the redundancy.

**Review by sdavtaker — APPROVED** [review]

**Pull request merged**